### PR TITLE
Port SemanticErrorAcceptance to the TCK

### PR DESCRIPTION
**Pull request by Mats-SX**
Adds several new error details:

- UndefinedVariable
- VariableTypeConflict
- RelationshipUniquenessViolation
- CreatingVarLength
- InvalidParameterUse
- InvalidClauseComposition
- FloatingPointOverflow
- PropertyAccessOnNonMap
- InvalidArgumentExpression
- InvalidUnicode
- NonConstantExpression
- NoSingleRelationshipType
- InvalidAggregation

### CODE REVIEW

File: tck/features/SemanticErrorAcceptance.feature [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+
+  Scenario: Failing when using `type()` on a node
+    When executing query:
+      """
+      MATCH (r)
+      RETURN type(r)
+      """
+    Then a SyntaxError should be raised at compile time: InvalidArgumentType
+
+  Scenario: Failing when a using `length()` on a node
```
Comment (petraselmer) [inline tck/features/SemanticErrorAcceptance.feature]: "Failing when a using `length()` on a node" -> "Failing when using `length()` on a node"

File: tck/features/SemanticErrorAcceptance.feature [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+  Scenario: Failing when deleting an integer expression
+    When executing query:
+      """
+      MATCH ()
+      DELETE 1 + 1
+      """
+    Then a SyntaxError should be raised at compile time: InvalidArgumentType
+
+  Scenario: Failing when using CREATE on already bound node
```
Comment (petraselmer) [inline tck/features/SemanticErrorAcceptance.feature]: "Failing when using CREATE on already bound node" -> "Failing when using CREATE on a node that is already bound"

**Comment by petraselmer** [comment]
@Mats-SX looks good; I just had a couple of comments regarding the scenario names...

**Comment by Mats-SX** [comment]
@petraselmer Fixed.

**Pull request merged**